### PR TITLE
feature toggle for debt and payments section.

### PR DESCRIPTION
**Pull request by tmitchellgcio**
…e for now.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#0000

## Screenshots
Screenshot 2021-12-29 111057
Screenshot 2021-12-29 111120

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/dashboard/components/benefit-payments-and-debts/DismissibleDebts.jsx
Comment (tmitchellgcio) [inline src/applications/personalization/dashboard/components/benefit-payments-and-debts/DismissibleDebts.jsx:10]: This is temporary. This will be hooked up to the VANotify APIs to dismiss the alert

**Comment by tmitchellgcio** [comment]
Small note on this. This is a working prototype for the debt and payments section hidden under the feature toggle. We intend to do user testing on this as well as hook up the top notification to VANotify before the feature is unhidden.

cc: @andaleliz

File: src/applications/personalization/dashboard/sass/user-profile.scss
Comment (timwright12) [inline src/applications/personalization/dashboard/sass/user-profile.scss:11]: Can we use design system utilities for some of these? https://design.va.gov/utilities/font-size

cc @k80bowman

Comment (k80bowman) [reply src/applications/personalization/dashboard/sass/user-profile.scss:11]: We have design system guidance on buttons here: https://design.va.gov/components/buttons
Please take a look, it looks like you are creating your own and we would like to keep button styles consistent.

**Review by k80bowman — CHANGES_REQUESTED** [review]

File: src/applications/personalization/dashboard/sass/user-profile.scss
Comment (tmitchellgcio) [reply src/applications/personalization/dashboard/sass/user-profile.scss:11]: Yes, but we don't have a design system utility that floats right or makes it transparent and I don't believe you can affect the hover with the design system utilities. The screenshot below is what I am building. I'm open to better ideas, but this is what my design team made. The X is the button. You can make a dismissible button on that but it doesn't look the same so I had to make my own. I can use some of those classes, but I can't remove that class altogether as it's what is making this transparent and float right.

Screenshot 2021-12-28 161141

Comment (tmitchellgcio) [reply src/applications/personalization/dashboard/sass/user-profile.scss:11]: also the design system utility has 4px not 5px for top padding so I can's use it for padding as there isn't one specific for it

Comment (tmitchellgcio) [reply src/applications/personalization/dashboard/sass/user-profile.scss:11]: Ok I updated what I could. It still needs the float right and the hover. Let me know if you think there is anything else I can do.

Comment (andaleliz) [reply src/applications/personalization/dashboard/sass/user-profile.scss:11]: Just FYI, this component is going to be reviewed by the design system council today.  

Also just a friendly reminder that this is a prototype for usability testing and not something we're proposing to release to Veterans on production without further refinements post-usability testing.

**Review by kelsonic — APPROVED** [review]
lgtm! 🎉

File: src/applications/personalization/dashboard/sass/user-profile.scss
Comment (k80bowman) [reply src/applications/personalization/dashboard/sass/user-profile.scss:11]: Oh good I'm glad to hear it will be reviewed by the design system council, that was going to be my next question. Thanks!

**Review by timwright12 — APPROVED** [review]
approving so I'm not the blocker here. I'm good whenever @k80bowman clear it through design system

**Review by adamwhitlock1 — APPROVED** [review]

File: src/applications/personalization/dashboard/sass/user-profile.scss
Comment (k80bowman) [reply src/applications/personalization/dashboard/sass/user-profile.scss:11]: @andaleliz does this need to be merged before it goes through the design system council?

**Review by jerekshoe — APPROVED** [review]

File: src/applications/personalization/dashboard/sass/user-profile.scss
Comment (andaleliz) [reply src/applications/personalization/dashboard/sass/user-profile.scss:11]: We don't need it to be merged for the design system council meeting, we have a static prototype for that. Our pilot study is Tuesday AM so ideally it'll be merged and ready for our team to review in time to make changes and prepare for that pilot. I'm totally unfamiliar with the timing of deploys and how long it takes to go to prod once merged.

Comment (tmitchellgcio) [reply src/applications/personalization/dashboard/sass/user-profile.scss:11]: @k80bowman we intend to test it with some staging users with specific profiles before we do user testing. Preferably, we would like to test it on staging today and have monday to make any last minute changes if we see something that may need an update so Tuesday goes smoothly. That's our intention currently

File: src/applications/check-in/api/local-mock-api/mocks/v2/feature-toggles/index.js
Comment (mdewey) [inline src/applications/check-in/api/local-mock-api/mocks/v2/feature-toggles/index.js:31]: Does this need to be added to a different team's local mock?

File: src/applications/check-in/utils/selectors/feature-toggles.js
Comment (mdewey) [inline src/applications/check-in/utils/selectors/feature-toggles.js:21]: Are you using this team's selectors?

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js
Comment (mdewey) [inline src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js:1]: To limit cross-application connections, you should create your own `generateFeatureToggles`

**Review by mdewey — CHANGES_REQUESTED** [review]
I added some notes about cross-app usage of code. Short term, I would say just duplicate what you need, but long term We should look at if this needs a refactor to more shared location

File: src/applications/check-in/utils/selectors/feature-toggles.js
Comment (tmitchellgcio) [reply src/applications/check-in/utils/selectors/feature-toggles.js:21]: Yes I am. Trying to limit duplicate code, however if this is not how it should be working then I can make my own generateFeatureToggles. I assumed duplicate code like that was frowned upon. Still new to the size of this organization. My old company had 4 developers.

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js
Comment (cohnjesse) [reply src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js:1]: @mdewey @timwright12 we should probably add a note about this to the documentation about feature toggles so it is more clear to new engineers, thoughts?

**Review by mdewey — APPROVED** [review]
LGTM. Good work, we can look at the refactor once this is in

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js
Comment (timwright12) [reply src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js]: @theodur is creating a script to check for things like this, we can have it run on changed files in a precommit hook

File: src/applications/personalization/dashboard/sass/user-profile.scss
Comment (k80bowman) [reply src/applications/personalization/dashboard/sass/user-profile.scss:11]: @tmitchellgcio thank you for explaining and updating what you could. After seeing the feature at the DSC meeting I understand better what it is you're doing here. Looks good.

**Review by k80bowman — APPROVED** [review]

**Pull request merged**
